savaslabs/savaslabs.github.io
# Issue #2368: Use Liquid variable for comment server URL

@kostajh This will allow us to use our local comment server by default when doing local development without having to update the URL in `main.js`.

I added a gulp plugin that allows us to prepend front matter dashes to our concatenated/uglified `main.js` file when it's placed in `assets`. It will then be handled by Jekyll during the `jekyll:build` process, where the Liquid variable `site.comment_server_url` will be processed. I've tested this in the local `gulp serve` process and the default `gulp` process, used in production, with successful results. Locally, I also changed what happens when one updates the main JS file - since we need Jekyll to process this file, not only will we need to run the `build:scripts` task, we'll need to run `build:jekyll:local` as well.

To test:

1. run `npm install` to install the new gulp plugin.
2. First, run `gulp`. This runs the default task, `build`, which builds the `_site` directory. This is what Travis will do when deploying out site. After this finishes (btw, you can remove `build:images` from the `default` task if you want to speed this up for testing purposes), check out `_site/assets/js/main.js`. Search for "http://comments.savaslabs.com" - you'll see that the variable has been correctly processed into our production comments URL.
3. Next, run `gulp serve`. This will use the dev config file when building the site. Once it's finished, look at `_site/assets/js/main.js` again. This time, search for "http://local.comments.savaslabs.com" - now it's using our local comments server URL.

Let me know if you have any suggestions for improving this. I also simplified some of the `clean` tasks.
